Add code prefetch support for DFS chase generator.

xyzsam opened the pull request:
This example is a bit more complex than the instruction pointer chase
CFG. Since we don't know for sure which way the branch will go, we can
either inject a prefetch for both targets at the very top, or we can
duplicate the function body for each side of the branch and prefetch
at the top of the branch. This second option would essentially be
equivalent to the instruction pointer chase CFG, so I've gone with the
first option. This can also be used as a model of uncertainty in the
control flow when injecting prefetches.

rpgolshan reviewed (commented):
Do we want to add prefetches for indirect branches? If not can you update the code to fail if indirect and prefetch is specified?

xyzsam commented on frontend/src/frontend/cfg_generator/common.py:
```diff
@@ -77,3 +77,34 @@ def _generate_cfg(self, functions: Dict[int, cfg_pb2.Function],
+        else:
+            raise ValueError('must specify one of function_id or code_block_id')
+        prefetch_inst.code_prefetch.degree = degree
+        prefetch_body = self._add_code_block()
```
I renamed prefetch_body to prefetch_block instead.

xyzsam commented:
> Do we want to add prefetches for indirect branches? If not can you update the code to fail if indirect and prefetch is specified?

Sorry, forgot about it. Added it, and also fixed the issue with useless code missing for indirect calls.
